csharp/src/Ice/ConnectionI.cs

```diff
@@ -1440,42 +1442,28 @@ internal ConnectionI(
         }
     }
 
-    /// <summary>Aborts the connection with a <see cref="ConnectionAbortedException" /> if the connection is active or
-    /// holding.</summary>
-    internal void idleCheck(TimeSpan idleTimeout, Action rescheduleTimer)
+    /// <summary>Aborts the connection with a <see cref="ConnectionAbortedException" /> if the connection is active and
+    /// does not receive any byte for some time. See the IdleCheckTransceiverDecorator.</summary>
+    internal void idleCheck(TimeSpan idleTimeout)
     {
         lock (this)
         {
-            if (_state == StateActive || _state == StateHolding)
```

Comment:
When the state is StateHolding, we're not reading off the network, so we should not "idle check". The difficulty is when we resume to StateActivate, we want to re-enable the idle check / read timer.